src/collection-view.js

```diff
@@ -125,11 +125,95 @@ const CollectionView = Backbone.View.extend({
     }
   },
 
-  // get the child view by model it holds, and remove it
-  _onCollectionRemove(model) {
-    const view = this.children.findByModel(model);
-    this.removeChildView(view);
-    this._checkEmpty();
+  // Handle collection update model removals
+  _onCollectionUpdate(collection, options) {
+    const changes = options.changes;
+    this._removeChildViews(changes.removed);
+  },
+
+  // Remove the child views and destroy them.
+  // This function also updates the indices of later views
+  // in the collection in order to keep the children in sync with the collection.
+  // "modelsOrViews" can be an array of mixed models or views that
+  // will be removed and destroyed from the CollectionView
+  _removeChildViews(modelsOrViews, {checkEmpty} = {}) {
+    const shouldCheckEmpty = checkEmpty !== false;
+
+    // Used to determine where to update the remaining
+    // sibling view indices after these views are removed.
+    const removedViews = this._getRemovedViews(modelsOrViews);
+
+    if (!removedViews.length) {
+      return;
+    }
+
+    this.children._updateLength();
+
+    if (this.sort) {
+      const greatestIndexedView = this._findGreatestIndexedView(removedViews);
```

Comment:
I'm thinking I should move this into `_updateIndices`
where that fuction will look like 
```
_updateIndices: function _updateIndices(views, increment, index) {
	    if (!this.sort) {
	      return;
	    }

      let view = views;
      if(_.isArray(view){
        view = this._findGreatestIndexedView(view);
      }

	    if (increment) {
	      // assign the index to the view
	      view._index = index;
	    }

	    // update the indexes of views after this one
	    this.children.each(function (laterView) {
	      if (laterView._index >= view._index) {
	        laterView._index += increment ? 1 : -1;
	      }
	    });
	  }
```
removing the need to do the check `_removeChildViews` and having the need in one place.